src/rgw/rgw_rest_s3.cc

```diff
@@ -2667,10 +2667,21 @@ int RGWCreateBucket_ObjStore_S3::get_params(optional_yield y)
 
 void RGWCreateBucket_ObjStore_S3::send_response()
 {
-  if (op_ret == -ERR_BUCKET_EXISTS)
-    op_ret = 0;
-  if (op_ret)
+  if (op_ret == -ERR_BUCKET_EXISTS) {
+    const auto eexist_override = s->cct->_conf.get_val<bool>("rgw_bucket_eexist_override");
+    if (! eexist_override) [[likely]] {
+      op_ret = 0;
+    } else {
+      /* n.b., EEXIST is mapped to HTTP error 409, BucketAlreadyExists, but ERR_BUCKET_EXISTS is not */
+      op_ret = -EEXIST;
+      /* https://repost.aws/knowledge-center/s3-error-bucket-already-exists */
+      s->err.message = "BucketAlreadyExists";
```

Comment:
according to https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html, the "Message" for BucketAlreadyExists is
> The requested bucket name is not available. The bucket namespace is shared by all users of the system. Specify a different name and try again.

we're already returning "BucketAlreadyExists" as the "Code" of the error response